Formula/ltex-ls.rb

```diff
@@ -0,0 +1,43 @@
+class LtexLs < Formula
+  desc "LSP for LanguageTool with support for Latex, Markdown and Others"
+  homepage "https://valentjn.github.io/ltex/"
+  url "https://github.com/valentjn/ltex-ls/archive/refs/tags/15.2.0.tar.gz"
+  sha256 "59209730cb9cda57756a5d52c6af459f026ca72c63488dee3cfd232e4cfbf70a"
+  license "MPL-2.0"
+  head "https://github.com/valentjn/ltex-ls.git", branch: "develop"
+
+  depends_on "maven" => :build
+  depends_on "python@3.10" => :build
+  depends_on "openjdk"
+
+  def install
+    ENV.prepend_path "PATH", Formula["python@3.10"].opt_libexec/"bin"
+    ENV["JAVA_HOME"] = Formula["openjdk"].opt_prefix
+    ENV["TMPDIR"] = buildpath
+    system Formula["python@3.10"].bin/"python3", "-u", "tools/createCompletionLists.py"
+    system "mvn", "-B", "-e", "-DskipTests", "package"
+    (buildpath/"build").mkpath
+    system "tar", "xzf", "target/ltex-ls-#{version}.tar.gz", "-C", "build"
+    # remove Windows files
+    rm Dir["build/ltex-ls-#{version}/bin/*.bat"]
+    bin.install Dir["build/ltex-ls-#{version}/bin/*"]
+    libexec.install Dir["build/ltex-ls-#{version}/*"]
+    bin.env_script_all_files libexec/"bin", Language::Java.overridable_java_home_env
```

Comment:
Let's use some whitespace to separate steps here. Additionally there is no need to specify the python path if it's in path anyway.
```suggestion
    ENV.prepend_path "PATH", Formula["python@3.10"].opt_libexec/"bin"
    ENV["JAVA_HOME"] = Formula["openjdk"].opt_prefix
    ENV["TMPDIR"] = buildpath

    system "python3", "-u", "tools/createCompletionLists.py"

    system "mvn", "-B", "-e", "-DskipTests", "package"

    mkdir "build" do
      system "tar", "xzf", "../target/ltex-ls-#{version}.tar.gz", "-C", "."

      # remove Windows files
      rm Dir["ltex-ls-#{version}/bin/*.bat"]
      bin.install Dir["ltex-ls-#{version}/bin/*"]
      libexec.install Dir["ltex-ls-#{version}/*"]
    end

    bin.env_script_all_files libexec/"bin", Language::Java.overridable_java_home_env
```